src/main/java/internet/store/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,151 @@
+        String query = "UPDATE orders SET deleted = true WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, orderId);
+            int updates = statement.executeUpdate();
+            return updates > 0;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Order # " + orderId + " was not deleted", ex);
+        }
+    }
+
+    @Override
+    public Optional<Order> get(Long orderId) {
+        String query = "SELECT order_id, user_id, product_id, product_name, price FROM orders"
+                + " JOIN orders_products USING(order_id)"
+                + " JOIN products ON (product_id = products.id)"
+                + " WHERE order_id = ? AND orders.deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, orderId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Long prodId = resultSet.getLong("product_id");
+                Long userId = resultSet.getLong("user_id");
+                Order order = new Order(userId);
+                order.setOrderId(orderId);
+                order.setProducts(retrieveProductData(resultSet));
+                return Optional.of(order);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("No order # " + orderId + " found", ex);
+        }
+        return Optional.empty();
+    }
+
+    private boolean deleteProductsFromOrder(Long orderId) {
+        String query = "DELETE FROM orders_products WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, orderId);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("products from order " + orderId
+                    + " were not deleted", ex);
+        }
+    }
+
+    private List<Order> getOrders(String query) {
+        List<Long> ordersId = new ArrayList<>();
+        List<Order> orders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                ordersId.add(resultSet.getLong("order_id"));
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("there are no orders", ex);
+        }
+        for (Long orderId : ordersId) {
+            orders.add(get(orderId).get());
+        }
```

Comment:
and fetch only products for order here?

Reply:
I'm using another method to add products - addProductsToOrder.